### ISSUE TITLE
How should I replace my own dataset for the experiment？

### ISSUE THREAD

**Issue by weican1103** [w4k2/torchosr#7]

**Comment by JKomorniczak** [comment]
You could write your own class that inherits from Dataset in Pytorch. As an example, you could use Madelon_base class in torchosr/data/base_datasets and provide data and targets objects (line 12) from your experimental data. 
With such dataset, you could use additional classes such as DataWrapper and OutlierDataset.

**Comment by weican1103** [comment]
Cool, I find your work to be very fascinating. I am currently in the process of revising a paper and need to add a comparative experiment. My dataset is in the .mat format (train.mat and test.mat), and it consists of 11 categories. For each category being considered as unknown, there is a corresponding set of train and test data; for instance, when the 0th category is unknown, there is one set of train and test data, and when the 1st category is unknown, there is another set of train and test data. I am interested in knowing if I can effectively integrate my own dataset within your framework. Secondly, I would like to understand if it's possible to appropriately customize the labels for known and unknown categories within the code.

**Comment by weican1103** [comment]
Cool, I find your work to be very fascinating. I am currently in the process of revising a paper and need to add a comparative experiment. My dataset is in the .mat format (train.mat and test.mat), and it consists of 11 categories. For each category being considered as unknown, there is a corresponding set of train and test data; for instance, when the 0th category is unknown, there is one set of train and test data, and when the 1st category is unknown, there is another set of train and test data. I am interested in knowing if I can effectively integrate my own dataset within your framework. Secondly, I would like to understand if it's possible to appropriately customize the labels for known and unknown categories within the code.



---- Replied Message ----
| From | ***@***.***> |
| Date | 03/14/2024 16:04 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [w4k2/torchosr] How should I replace my own dataset for the experiment？ (Issue #7) |

Closed #7 as completed.

**Comment by JKomorniczak** [comment]
This may require the use (and possibly a slight modification) of OutlierDataset class, where you would pass subsets containing known and unknown objects separately. The OutlierDataset creates appropriate labels for this scenario by re-assigning successive labels to *n* known classes and a label *n+1* to all unknown objects.
